### DISCUSSION TITLE
[RFC] Modal support trigger

### DISCUSSION THREAD

**Discussion by daledelv** [ant-design/ant-design#46766] [RFCs]
### What problem does this feature solve?

**I hope to provide a props called "trigger" for Modal, similar to [ModalForm](https://procomponents.ant.design/components/modal-form#modalformdrawerform-demo-modal-form) in procomponent.**
Because modal components are read-only in daily work, I don't want to maintain these redundant state, and I hope these jsx codes can be aggregated together to facilitate daily maintenance.
In most scenarios, I think Modal components should have the same experience as Tooltip components.

### What is expected?

```tsx
<Modal
  title="操作日志"
  footer={null}
  trigger={
    <Button type="primary">
      日志
    </Button>
  }
>
  <Timeline>
    <Timeline.Item></Timeline.Item>
  </Timeline>
</Modal>
```

### What is actually happening?

```tsx
const [log_visable, set_log_visable] = useState(false);
async function seeLog(id: number) {
  set_log_visable(true);
}
<>
  <Button type="primary" onClick={() => seeLog(record.id)}>
    日志
  </Button>
  <Modal title="操作日志" open={log_visable} onCancel={() => set_log_visable(false)} footer={null}>
    <Timeline>
      <Timeline.Item></Timeline.Item>
    </Timeline>
  </Modal>
</>
```

**Comment by afc163** [comment]
Same idea as https://juejin.cn/post/7265833406226251812

**Reply by Wxh16144** [reply]
`trigger` 这个感觉还自己做 HOC 好一些，但是模态框的非受控真可以加一个 `defaultOpen` 属性

**Comment by afc163** [comment]
One of reason I don't prefer `trigger` property is that we usually treat `children` as trigger in other antd components like Tooltip/Popover/Popconfirm.

```jsx
<Popover>
  <button>trigger is here!<button>
</Popover>
```

**Comment by zombieJ** [comment]
antd positioning is an atomic component library. API will be expanded when the functional requirement cannot be satisfied with HOC. You can use ProComponent which has the prop.

**Comment by Wxh16144** [comment]
Sounds like bad news, no plans for trigger support for now!
